server/webapp/WEB-INF/rails.new/app/assets/javascripts/vsm_analytics.js

```diff
@@ -51,25 +51,50 @@
     }
   };
 
-  var showAnalytics = function showAnalytics(options) {
-    var div = document.createElement("div");
-
+  function defineHandlers(chartId) {
     PluginEndpointRequestHandler.defineLinkHandler();
+    const models    = {};
+    models[chartId] = {
+      fetch: function (url, handler) {
+        const splitURL = url.split('?');
+        const search   = splitURL[1];
+        const jsonData = JSON.parse('{"' + decodeURIComponent(search).replace(/"/g, '\\"').replace(/&/g, '","').replace(/=/g, '":"') + '"}');
+        $j.ajax({
+          url:      splitURL[0],
+          type:     "POST",
+          dataType: "json",
+          data:     jsonData
+        }).done((r) => {
+          handler(r.data, null);
+        });
+      }
+    };
+
+    PluginEndpointRequestHandler.defineFetchAnalyticsHandler(models);
     PluginEndpoint.ensure("v1");
+  }
+
+  var showAnalytics = function showAnalytics(options) {
```

Comment:
No need to declare a `var` once you've declared a named `function`.

```
function showAnalytics(options) {
  ...
}
```

Is sufficient.